pkg/lib/cockpit-components-file-autocomplete.jsx

```diff
@@ -59,13 +66,23 @@ var FileAutoComplete = React.createClass({
         if (value && value.indexOf("/") !== 0)
             value = "/" + value;
 
-        if (!this.updateIfDirectoryChanged(value));
-            this.filterFiles(value);
-
-        this.setState({
+        var stateUpdate = {
             value: value,
+        };
+
+        if (!this.updateIfDirectoryChanged(value))
+            Object.assign(stateUpdate, this.filterFiles(value));
```

Comment:
We can't do this. This widget is used in places where object.assign is not available.